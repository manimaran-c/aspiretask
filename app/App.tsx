import React, {Fragment} from 'react';
import {SafeAreaView, StatusBar, StyleSheet} from 'react-native';
import Navigator from 'app/navigation';
import AppStyles from './config/AppStyles';
import {ActivityIndicator} from 'react-native';
import {Provider} from 'react-redux';
import {PersistGate} from 'redux-persist/es/integration/react';
import configureStore from 'app/store';

const {persistor, store} = configureStore();

const App = () => {
  return (
    <Provider store={store}>
      <PersistGate loading={<ActivityIndicator />} persistor={persistor}>
        <Fragment>
          <SafeAreaView style={styles.topColor} />
          <SafeAreaView style={styles.container}>
            <StatusBar barStyle="light-content" />
            <Navigator />
          </SafeAreaView>
        </Fragment>
      </PersistGate>
    </Provider>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: AppStyles.colors.COLOR_WHITE,
  },
  topColor: {
    flex: 0,

    backgroundColor: AppStyles.colors.COLOR_BG,
  },
});

export default App;
