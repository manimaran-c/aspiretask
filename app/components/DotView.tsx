import React from 'react';
import {StyleSheet, TextProps, View} from 'react-native';
import AppStyles from 'app/config/AppStyles';

interface Props extends TextProps {
  count?: number;
}

const DotView: React.FC<Props> = ({count = 4, ...props}: Props) => {
  var elements = [];
  for (var i = 0; i < count; i++) {
    elements.push(
      <View
        style={[styles.dotView, i === 0 && styles.zeroMarginLeft]}
        key={i.toString()}
      />,
    );
  }

  return <View style={[styles.dotContainer, props.style]}>{elements}</View>;
};

const styles = StyleSheet.create({
  dotContainer: {
    flexDirection: 'row',
    marginLeft: 24,
  },
  dotView: {
    width: 8,
    height: 8,
    borderRadius: 8,
    backgroundColor: AppStyles.colors.COLOR_WHITE,
    marginLeft: 6,
  },
  zeroMarginLeft: {
    marginLeft: 0,
  },
});
export default DotView;
