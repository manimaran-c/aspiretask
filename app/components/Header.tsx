import AppStyles from 'app/config/AppStyles';
import Images from 'app/config/Images';
import NavigationService from 'app/navigation/NavigationService';
import React from 'react';
import {View, TouchableOpacity, StyleSheet, Image} from 'react-native';

export interface IHeaderProps {
  showBack?: boolean;
}

const Header: React.FC<IHeaderProps> = ({showBack = false}) => {
  return (
    <View style={[styles.header]}>
      {showBack ? (
        <TouchableOpacity
          onPress={() => {
            NavigationService.goBack();
          }}
          style={styles.backView}
        >
          <Image source={Images.back} style={styles.backImg} />
        </TouchableOpacity>
      ) : (
        <>
          <View />
        </>
      )}
      <Images.logo width={26} height={26} />
    </View>
  );
};

const styles = StyleSheet.create({
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: AppStyles.colors.COLOR_BG,
    height: 56,
    justifyContent: 'space-between',
    paddingHorizontal: 24,
  },
  backView: {
    width: 45,
    height: 45,
    justifyContent: 'center',
  },
  backImg: {
    width: 20,
    height: 20,
  },
});
export default Header;
