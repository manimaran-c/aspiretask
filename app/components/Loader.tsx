import React from 'react';
import {StyleSheet, View, Modal, ActivityIndicator} from 'react-native';
import {useSelector} from 'react-redux';
import IState from 'app/models/reducers';

const Loader: React.FC = () => {
  const isLoading = useSelector(
    (state: IState) => state.loadingReducer.isLoading,
  );

  return (
    <Modal
      transparent={true}
      animationType={'none'}
      visible={isLoading}
      style={styles.zIndex}
      onRequestClose={() => {}}
    >
      <View style={styles.modalBackground}>
        <View style={styles.activityIndicatorWrapper}>
          <ActivityIndicator animating={isLoading} color="black" />
        </View>
      </View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  modalBackground: {
    flex: 1,
    alignItems: 'center',
    flexDirection: 'column',
    justifyContent: 'space-around',
    backgroundColor: '#rgba(0, 0, 0, 0.5)',
    zIndex: 1000,
  },
  activityIndicatorWrapper: {
    backgroundColor: '#FFFFFF',
    height: 100,
    width: 100,
    borderRadius: 10,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  zIndex: {
    zIndex: 1100,
  },
});

export default Loader;
