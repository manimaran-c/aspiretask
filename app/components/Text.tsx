/**
 * Common Text Component
 */
import React from 'react';
import {Text as RText, StyleSheet, TextProps} from 'react-native';
import AppStyles from 'app/config/AppStyles';

interface Props extends TextProps {
  children?: any;
}

const Text: React.FC<Props> = ({children, ...props}: Props) => {
  return <RText style={[styles.text, props.style]}>{children}</RText>;
};

const styles = StyleSheet.create({
  text: {
    fontSize: 14,
    fontFamily: AppStyles.fonts.FONT_REGULAR,
    color: AppStyles.colors.COLOR_WHITE,
  },
});
export default Text;
