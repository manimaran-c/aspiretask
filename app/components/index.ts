export {default as Text} from './Text';
export {default as Header} from './Header';
export {default as DotView} from './DotView';
export {default as Loader} from './Loader';
