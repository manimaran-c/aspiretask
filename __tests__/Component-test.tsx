import 'react-native';
import React from 'react';
import {DotView, Header, Text} from 'app/components';
import renderer from 'react-test-renderer';

it('text renders correctly across screens', () => {
  const tree = renderer.create(<Text />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('header renders correctly across screens', () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('dotView renders correctly across screens', () => {
  const tree = renderer.create(<DotView />).toJSON();
  expect(tree).toMatchSnapshot();
});
