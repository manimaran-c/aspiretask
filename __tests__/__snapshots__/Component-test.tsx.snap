// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dotView renders correctly across screens 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "marginLeft": 24,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 8,
          "height": 8,
          "marginLeft": 6,
          "width": 8,
        },
        Object {
          "marginLeft": 0,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 8,
          "height": 8,
          "marginLeft": 6,
          "width": 8,
        },
        false,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 8,
          "height": 8,
          "marginLeft": 6,
          "width": 8,
        },
        false,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 8,
          "height": 8,
          "marginLeft": 6,
          "width": 8,
        },
        false,
      ]
    }
  />
</View>
`;

exports[`header renders correctly across screens 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#0C365A",
        "flexDirection": "row",
        "height": 56,
        "justifyContent": "space-between",
        "paddingHorizontal": 24,
      },
    ]
  }
>
  <View />
  <SvgMock
    height={26}
    width={26}
  />
</View>
`;

exports[`text renders correctly across screens 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#FFFFFF",
        "fontFamily": "Metropolis-Regular",
        "fontSize": 14,
      },
      undefined,
    ]
  }
/>
`;
